qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,824 @@
+        valid_only: bool = True,
+        group: str = "default",
+        cutoff_date: datetime = None,
+    ) -> Union[int, float, complex]:
+        """
+        Retrieves the value of a parameter.
+
+        Parameters may be linked. get_parameter_value does the following steps:
+        1) Retrieve the parameter object corresponding to (param, qubits, schedule)
+        2) The values of this parameter may be stored under another schedule since
+           schedules can share parameters. To deal with this, a list of candidate keys
+           is created internally based on the current configuration.
+        3) Look for candidate parameter values under the candidate keys.
+        4) Filter the candidate parameter values according to their date (up until the
+           cutoff_date), validity and calibration group.
+        5) Return the most recent parameter.
+
+        Args:
+            param: The parameter or the name of the parameter for which to get the parameter value.
+            qubits: The qubits for which to get the value of the parameter.
+            schedule: The schedule or its name for which to get the parameter value.
+            valid_only: Use only parameters marked as valid.
+            group: The calibration group from which to draw the parameters.
+                If not specified this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            value: The value of the parameter.
+
+        Raises:
+            CalibrationError:
+                - If there is no parameter value for the given parameter name and pulse channel.
+        """
+
+        # 1) Identify the parameter object.
+        param_name = param.name if isinstance(param, Parameter) else param
+        sched_name = schedule.name if isinstance(schedule, (Schedule, ScheduleBlock)) else schedule
+
+        param = self.calibration_parameter(param_name, qubits, sched_name)
+
+        # 2) Get a list of candidate keys restricted to the qubits of interest.
+        candidate_keys = []
+        for key in self._parameter_map_r[param]:
+            candidate_keys.append(ParameterKey(key.parameter, qubits, key.schedule))
+
+        # 3) Loop though the candidate keys to candidate values
+        candidates = []
+        for key in candidate_keys:
+            if key in self._params:
+                candidates += self._params[key]
+
+        # If no candidate parameter values were found look for default parameters
+        # i.e. parameters that do not specify a qubit.
+        if len(candidates) == 0:
+            for key in candidate_keys:
+                if ParameterKey(key.parameter, None, key.schedule) in self._params:
```

Comment:
This is minor comment. I think `None` means the value is not assigned. However, in this case, actually the index is assigned to all qubits. This is slightly different from what `None` indicates. For example, if we have a parameter that doesn't have qubits, this should be `qubits=None` and this doesn't mean the parameter is associated with all qubits.

Is it possible to explicitly define the index is associated with all qubits? For example,
```python
AllQubits = TypeVar('AllQubits')

if index is AllQubits:
   do something
```

I think using `None` is also fine, but maybe having explicit notation would make the parameter table more readable.

Reply:
In 320b377 the None has been replaced with an empty tuple. Defining a new type var is overkill here.